improver_tests/cube_combiner/test_Combine.py

```diff
@@ -0,0 +1,131 @@
+
+
+@pytest.fixture(name="realization_cubes")
+def realization_cubes_fixture() -> CubeList:
+    """Set up a single realization cube in parameter space"""
+    realizations = [0, 1, 2, 3]
+    data = np.ones((len(realizations), 2, 2), dtype=np.float32)
+    times = [datetime(2017, 11, 10, hour) for hour in [4, 5, 6]]
+    cubes = CubeList()
+    for time in times:
+        cubes.append(
+            set_up_variable_cube(
+                data,
+                realizations=realizations,
+                spatial_grid="equalarea",
+                time=time,
+                frt=datetime(2017, 11, 10, 1),
+            )
+        )
+    cube = cubes.merge_cube()
+    return CubeList(cube.slices_over("realization"))
+
+
+@pytest.mark.parametrize("broadcast_to_threshold", (False, True))
+@pytest.mark.parametrize("minimum_realizations", (None, 1))
+@pytest.mark.parametrize(
+    "operation, expected_instance", (("+", CubeCombiner), ("*", CubeMultiplier)),
+)
+def test_init(
+    operation, expected_instance, minimum_realizations, broadcast_to_threshold
+):
+    """Ensure the class initialises as expected"""
+    result = Combine(
+        operation,
+        new_name="name",
+        minimum_realizations=minimum_realizations,
+        broadcast_to_threshold=broadcast_to_threshold,
+    )
+    assert isinstance(result.plugin, expected_instance)
+    assert result.new_name == "name"
+    assert result.minimum_realizations == minimum_realizations
+    assert result.broadcast_to_threshold == broadcast_to_threshold
+    if broadcast_to_threshold and isinstance(result.plugin, CubeMultiplier):
+        assert result.plugin.broadcast_to_threshold == broadcast_to_threshold
+
+
+@pytest.mark.parametrize("short_realizations", [0, 1, 2, 3])
+def test_filtering_realizations(realization_cubes, short_realizations):
+    """Run Combine with minimum_realizations and a realization time series where 0 or more are
+    short of the final time step"""
+    if short_realizations == 0:
+        cubes = realization_cubes
+        expected_realization_points = [0, 1, 2, 3]
+    else:
+        cubes = CubeList(realization_cubes[:-short_realizations])
+        cubes.append(realization_cubes[-short_realizations][:-1])
+        expected_realization_points = [0, 1, 2, 3][:-short_realizations]
+    result = Combine(
+        "+", new_name="name", minimum_realizations=1, broadcast_to_threshold=False,
+    )(cubes)
+    assert isinstance(result, Cube)
+    assert np.allclose(result.coord("realization").points, expected_realization_points)
+    assert np.allclose(result.data, 3)
+
+
+@pytest.mark.parametrize(
+    "minimum_realizations, error_class, msg",
+    (
+        (0, ValueError, "Minimum realizations must be at least 1, not 0"),
+        ("0", ValueError, "Minimum realizations must be at least 1, not 0"),
+        (-1, ValueError, "Minimum realizations must be at least 1, not -1"),
+        (5, ValueError, "After filtering, number of realizations 4 is less than 5"),
```

Comment:
Should this situation actually result in a different error message to this? In this particular situation, the `minimum_realizations` is greater than the number of realizations available pre-filtering, rather than post-filtering. Conceptually, this could be done by making the "if self.minimum_realizations < 1" check into an "if self.minimum_realizations within the valid range check" e.g.
```
if self.minimum_realizations < 1 and all([len(c.coord("realization").points) > self.minimum_realizations for c in cubes])
```
Alternatively, it could just be added as a separate pre-filtering check.